runtime/js-compute-runtime/builtins/cache-simple.cpp

```diff
@@ -617,27 +563,20 @@ bool SimpleCache::set(JSContext *cx, unsigned argc, JS::Value *vp) {
       return false;
     }
 
-    // The cache APIs require the length to be known upfront, we don't know the length of a
-    // stream upfront, which means the caller will need to supply the information explicitly for us.
-    if (!args.hasDefined(3)) {
-      JS_ReportErrorASCII(cx, "SimpleCache.set: length parameter is required when the value "
-                              "parameter is a ReadableStream. The length of the stream needs to be "
-                              "known before inserting into the cache.");
-      return false;
-    }
-
-    JS::HandleValue length_val = args.get(3);
-    double number;
-    if (!JS::ToNumber(cx, length_val, &number)) {
-      return false;
-    }
-    if (number < 0 || std::isnan(number) || std::isinf(number)) {
-      JS_ReportErrorASCII(
-          cx, "SimpleCache.set: length parameter is an invalid value, only positive numbers can "
-              "be used for length values.");
-      return false;
+    if (args.hasDefined(3)) {
```

Comment:
The length parameter is not actually required, what happens if the length is supplied is that any clients that begin streaming the item’s contents before it is completely provided will see the promised length which allows them to, for example, use content-length instead of transfer-encoding: chunked if the item is used as the body of a Request or Response